src/transaction.rs

```diff
@@ -7,24 +7,31 @@ use plan::{Condition, Payment, Plan};
 use signature::{KeyPair, KeyPairUtil, PublicKey, Signature, SignatureUtil};
 
 #[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
-pub struct Transaction {
+pub struct Signed {
     pub from: PublicKey,
-    pub plan: Plan,
     pub tokens: i64,
     pub last_id: Hash,
+    pub plan: Plan,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+pub struct Transaction {
     pub sig: Signature,
+    pub data: Signed,
 }
 
 impl Transaction {
     pub fn new(from_keypair: &KeyPair, to: PublicKey, tokens: i64, last_id: Hash) -> Self {
         let from = from_keypair.pubkey();
         let plan = Plan::Pay(Payment { tokens, to });
         let mut tr = Transaction {
-            from,
-            plan,
-            tokens,
-            last_id,
             sig: Signature::default(),
+            data: Signed {
+                from,
```

Comment:
`from` probably shouldn't be in there, eh?

Reply:
Fixing this here: https://github.com/solana-labs/solana/pull/100/commits/6fec8fad573623a23b2cf6a6cd66a9f80e542be1